### Hospital Patient Flow & Care Analytics

You are a data analyst at a hospital. Management wants insights on:

- patient volume (daily/weekly)
- waiting times and length of stay
- readmissions
- department performance
- identifying risk/high-priority cases



Dataset 
A) Patient records (rows = patients)

Columns description:

- Age
- Department code (0=Emergency, 1=Outpatient, 2=Surgery, 3=Pediatrics)
- Severity score (1–5)
- Waiting time (minutes)
- Length of stay (hours)
- Readmitted within 30 days? (0=No, 1=Yes)
- Treatment cost 

In [3]:
departments = np.array(["Emergency", "Outpatient", "Surgery", "Pediatrics"])

patients = np.array([
    [25, 0, 4, 60, 10, 0, 350],
    [60, 2, 5, 30, 48, 1, 4200],
    [40, 1, 2, 20,  2, 0, 150],
    [10, 3, 3, 15,  6, 0, 200],
    [72, 0, 5, 90, 24, 1, 1800],
    [33, 1, 1, 10,  1, 0, 80],
    [55, 2, 4, 45, 36, 0, 3500],
    [28, 0, 3, 50,  8, 0, 500],
    [65, 2, 5, 25, 72, 1, 9000],
    [18, 3, 2, 12,  3, 0, 120],
    [49, 1, 3, 35,  4, 0, 220],
    [80, 0, 4, 110, 30, 1, 2500],
])

days = np.array(["Mon","Tue","Wed","Thu","Fri","Sat","Sun"])

daily_visits = np.array([
    [30, 50, 10, 20],
    [35, 55, 12, 18],
    [40, 60, 15, 22],
    [38, 58, 14, 25],
    [45, 65, 18, 30],
    [50, 40, 20, 15],
    [55, 35, 22, 10],
])

### Part 1 — Patient Summary KPIs

1. Total number of patients in the patients dataset.
2. Average age of patients.
3. Average waiting time (minutes).
4. Average length of stay (hours).
5. Total treatment cost for all patients.

### Part 2 — Weekly Flow (Daily visits table)

6. Total visits in the week (all departments combined).
7. Total visits per department for the week.
8. Which day had the highest total visits?
9. Which department had the highest visits on Saturday?

### Part 3 — Department Performance (Patients table)

10. Count how many patients are in each department (Emergency/Outpatient/Surgery/Pediatrics).
11. Compute average waiting time per department.
12. Compute average cost per department.
13. Which department has the highest average waiting time?

In [3]:
import numpy as np
print("successfully installed")


successfully installed


In [19]:
departments = np.array(["Emergency", "Outpatient", "Surgery", "Pediatrics"])

patients = np.array([
#   age  DC SS WT  LS RW  TC
    [25, 0, 4, 60, 10, 0, 350],
    [60, 2, 5, 30, 48, 1, 4200],
    [40, 1, 2, 20,  2, 0, 150],
    [10, 3, 3, 15,  6, 0, 200],
    [72, 0, 5, 90, 24, 1, 1800],
    [33, 1, 1, 10,  1, 0, 80],
    [55, 2, 4, 45, 36, 0, 3500],
    [28, 0, 3, 50,  8, 0, 500],
    [65, 2, 5, 25, 72, 1, 9000],
    [18, 3, 2, 12,  3, 0, 120],
    [49, 1, 3, 35,  4, 0, 220],
    [80, 0, 4, 110, 30, 1, 2500],
])



## part one

In [8]:
total_patients = np.shape (patients)
print("The total patient is: ", total_patients)

The total patient is:  (12, 7)


### 12 rows 7 columns 
### 12 patient 7 variables 
### the hospital has 12 patient in total

In [11]:
average_age = np.average(patients[:,0])
print("The average age of patient is: ", average_age)

The average age of patient is:  44.583333333333336


### the  average age of patient in the hospital is 45

In [13]:
average_waiting_time= np.average(patients[:,3])
print("The average waiting time  of patient is: ", average_waiting_time)

The average waiting time  of patient is:  41.833333333333336


### on average,patient wait about 42 minute before seen.

In [22]:
total_treatment= np.sum(patients[:,6])
print("The total treatment  of patient is: ", total_treatment)

The total treatment  of patient is:  22620


###  the total treatment cost for all patient is #22620

In [14]:
average_length_of_stay= np.average(patients[:,4])
print("The average length of stay of  patient is: ",average_length_of_stay)

The average length of stay of  patient is:  20.333333333333332


### The average length of stay for patient is roughly 20hrs

### part two

In [14]:
daily_visits = np.array([
    [30, 50, 10, 20],
    [35, 55, 12, 18],
    [40, 60, 15, 22],
    [38, 58, 14, 25],
    [45, 65, 18, 30],
    [50, 40, 20, 15],
    [55, 35, 22, 10],
])
departments = np.array(["Emergency", "Outpatient", "Surgery", "Pediatrics"])
days = np.array(["Mon","Tue","Wed","Thu","Fri","Sat","Sun"]) 

In [15]:
total_visit= np.sum(daily_visits )
print("The total visit per department  for the week is: ", total_visit )

The total visit per department  for the week is:  907


### Across all department the total visit is 907 per week

In [17]:
total_visit= np.sum(daily_visits,axis=0)


In [18]:
print(departments[0],total_visit[0])
print(departments[1],total_visit[1])
print(departments[2],total_visit[2])
print(departments[3],total_visit[3])

Emergency 293
Outpatient 363
Surgery 111
Pediatrics 140


### surgery has the higest visit  per week followed by  pediatrics followed by  outpatient,lastly emergency

In [19]:
total_per_day= np.sum(daily_visits,axis=1)
print("the total visit  per day for each department is: ",total_per_day)

the total visit  per day for each department is:  [110 120 137 135 158 125 122]


In [23]:
max_day_index=np.argmax(total_per_day)
print("day with the higest visit:")
str(days[max_day_index])

day with the higest total visit:


'Fri'

### friday has the highest visit in the weeek which is also the buiset day in the weeek

In [76]:
print(departments[0],"is the highest department visit on saturday" )

Emergency is the highest department visit on saturday


### emergency department is always the busiet on saturday

### part three

In [4]:
departments = np.array(["Emergency", "Outpatient", "Surgery", "Pediatrics"])

patients = np.array([
#   age  DC SS WT  LS RW  TC
    [25, 0, 4, 60, 10, 0, 350],
    [60, 2, 5, 30, 48, 1, 4200],
    [40, 1, 2, 20,  2, 0, 150],
    [10, 3, 3, 15,  6, 0, 200],
    [72, 0, 5, 90, 24, 1, 1800],
    [33, 1, 1, 10,  1, 0, 80],
    [55, 2, 4, 45, 36, 0, 3500],
    [28, 0, 3, 50,  8, 0, 500],
    [65, 2, 5, 25, 72, 1, 9000],
    [18, 3, 2, 12,  3, 0, 120],
    [49, 1, 3, 35,  4, 0, 220],
    [80, 0, 4, 110, 30, 1, 2500],
])

In [10]:
num_department = [np.sum(patients[:,1] == i) for i in range (4)]
Department = ["Emergency", "Outpatient", "Surgery", "Pediatrics"]
j = 0
for i in num_department:
    print({Department[j]:int(i)})
    j+=1

{'Emergency': 4}
{'Outpatient': 3}
{'Surgery': 3}
{'Pediatrics': 2}


### Emergency 4 patient
### Outpatient and Surgery 3 patient 
### pediatrics 2 patient


In [9]:
avg_waiting_time =[np.mean(patients[patients[:,1]==i, 3])for i in range(4)]
print(avg_waiting_time)

[np.float64(77.5), np.float64(21.666666666666668), np.float64(33.333333333333336), np.float64(13.5)]


### Emergency has the  highest average waiting 

In [14]:
avg_cost= [np.mean(patients[patients[:,1] == i,6])for i in range(4)]
for i in range(4):
    print(departments[i],"average cost:",avg_cost[i])

Emergency average cost: 1287.5
Outpatient average cost: 150.0
Surgery average cost: 5566.666666666667
Pediatrics average cost: 160.0


In [19]:
highest_avg_waiting_time = [np.mean(patients[patients[:,1] == i,3]) for i in range (4)]
print(departments[np.argmax(highest_avg_waiting_time)])

Emergency


## Insight

- Patient Volume:
### Daily and weekly patient visits show Pediatrics and Emergency are the busiest, indicating higher demand and need for adequate staffing.
- Waiting Time:
### Emergency has the highest average waiting time, suggesting workflow improvements are needed.
- Readmission:
### Some patients return for care, highlighting the need for better follow-up.
- Department Performance:
### Efficiency varies by department based on load, waiting time, and cost.
- Risk / High-Priority Cases:
### patients with higher risk level  must be prioritized for timely treatment.